### ISSUE TITLE
Validate RKNN INT8 inference outputs on RK3588 and RK3588S

### ISSUE THREAD

**Issue by glenn-jocher** [ultralytics/ultralytics#25497]
## Summary

Track real-device RKNN INT8 inference correctness after successful Ultralytics export/conversion.

A Discourse report shows a YOLO26n model that works on a RADXA 5B before quantization but produces zero confidence/class outputs, incorrect coordinates, or occasional segmentation faults after INT8 RKNN conversion. Earlier RKNN issues primarily tracked adding INT8 export support and have been closed.

Ultralytics 8.4.110 now validates RKNN Toolkit conversion for every current YOLO task, but successful conversion alone does not prove correct NPU runtime outputs.

## Reproduction target

Use the official exporter with a representative detection model and calibration dataset, for example:

```bash
yolo export model=yolo26n.pt format=rknn name=rk3588 quantize=8 data=coco8.yaml
```

Run the exported model on RK3588 and/or RK3588S hardware with a runtime version compatible with the RKNN Toolkit version used for export.

Record:

- Device and operating-system image
- `rknn-toolkit2` and RKNN runtime versions
- Export command and calibration dataset
- Input preprocessing, color order, dtype, and normalization
- Raw FP16 and INT8 outputs for the same inputs
- Postprocessed detections and validation accuracy

## Expected behavior

INT8 inference produces finite, nonzero confidence/class outputs and geometrically valid boxes, with an explainable accuracy difference from the floating-point RKNN model.

## Acceptance criteria

- A reproducible RK3588/RK3588S validation procedure is documented.
- FP16 and INT8 exports are run on the same device and inputs.
- INT8 output tensors and postprocessed results are checked for correctness, not only successful model loading.
- If the report reproduces, the root cause is fixed at the owning export, calibration, metadata, preprocessing, or runtime boundary.
- A durable device/runtime validation artifact or documented manual release gate prevents conversion-only testing from being mistaken for runtime validation.
- The originating Discourse user is updated with the verified command and result.

## References

- Discourse report: https://community.ultralytics.com/t/quantization-issue-with-yolo26n-rknn-on-radxa5b-outputs-scores-0-0-class-id-0-0/1799/4
- All-task RKNN export support: https://github.com/ultralytics/ultralytics/pull/25490
- Previous INT8 export tracking: https://github.com/ultralytics/ultralytics/issues/24613
- RKNN documentation: https://docs.ultralytics.com/integrations/rockchip-rknn/

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for opening this RKNN INT8 validation issue 🚀! This is an automated response, and an Ultralytics engineer will also assist soon.

For this 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example), including the exact export and runtime commands, so we can investigate the behavior reliably.

To help validate RKNN INT8 inference on RK3588/RK3588S, please include:

- Device model, operating-system image, and hardware details
- `rknn-toolkit2` and RKNN runtime versions
- Ultralytics package version and export command
- Calibration dataset and representative input images
- Input preprocessing details, including color order, dtype, and normalization
- FP16 and INT8 raw output tensors for identical inputs
- Postprocessed detections, confidence/class outputs, and validation accuracy
- Complete logs, including any warnings, errors, or segmentation faults
- A comparison between FP16 and INT8 inference on the same device and inputs

Please also verify that you are using the latest `ultralytics` package and following the [RKNN integration documentation](https://docs.ultralytics.com/integrations/rockchip-rknn/).

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

For additional context, please review the [Discourse report](https://community.ultralytics.com/t/quantization-issue-with-yolo26n-rknn-on-radxa5b-outputs-scores-0-0-class-id-0-0/1799/4), [all-task RKNN export support](https://github.com/ultralytics/ultralytics/pull/25490), [previous INT8 export tracking](https://github.com/ultralytics/ultralytics/issues/24613), and the [RKNN documentation](https://docs.ultralytics.com/integrations/rockchip-rknn/).
